spark/src/main/java/org/apache/iceberg/spark/SparkStructLike.java

```diff
@@ -19,11 +19,12 @@
 
 package org.apache.iceberg.spark;
 
+import java.io.Serializable;
 import org.apache.iceberg.StructLike;
 import org.apache.iceberg.types.Types;
 import org.apache.spark.sql.Row;
 
-public class SparkStructLike implements StructLike {
+public class SparkStructLike implements StructLike, Serializable {
```

Comment:
Required for the SparkDataFile SerDe

Reply:
Shouldn't `SparkDataFile` be serialized as an internal row and not using Java serialization?

Reply:
Okay, I see that it is because that is being collected inside a FileScanTask.

Reply:
Where is `SparkStructLike` serialized?

Reply:
Same place, SparkDataFile. It declares the wrapped partition as SparkStructLike

Reply:
` private final SparkStructLike wrappedPartition;`

Reply:
Got it. But are those really serialized back? The filtering currently wraps and filters, but the original `Row` objects are returned to Spark. I don't think we are serializing `SparkDataFile` if we construct tasks on the driver.

Reply:
Yeah they only are serialized going out,
We create a SparkDataFile, pack it into a FileScanTask which becomes a CombinedScanTask which we then make into a ReadTask, which ends up having to be serialized. (In Spark 3, it's slightly different in Spark2)

Reply:
So it's not that we are doing an explicit parallelizing or collecting, but when we make our Spark Partitions/InputSplits, those need to be serializable.

Reply:
Ok, so it is not for distributed planning as such, it is needed after the planning is done since our DataFiles are actually SparkDataFiles under the hood?

Reply:
Yep, it's needed for Distributed Execution, not distributed planning